sympy/printing/llvmjitcode.py

```diff
@@ -0,0 +1,192 @@
+        fn = ll.Function(self.module, fn_type, "pow")
+        return self.builder.call(fn, [base0, exp0], "pow")
+
+    def _print_Mul(self, expr):
+        nodes = [self._print(a) for a in expr.args]
+        e = nodes[0]
+        for node in nodes[1:]:
+            e = self.builder.fmul(e, node)
+        return e
+
+    def _print_Add(self, expr):
+        nodes = [self._print(a) for a in expr.args]
+        e = nodes[0]
+        for node in nodes[1:]:
+            e = self.builder.fadd(e, node)
+        return e
+
+    # TODO - assumes all called functions take one double precision argument.
+    #        Should have a list of math library functions to validate this.
+    def _print_Function(self, expr):
+        name = expr.func.__name__
+        e0 = self._print(expr.args[0])
+        fn_type = ll.FunctionType(self.fp_type, [self.fp_type])
+        fn = ll.Function(self.module, fn_type, name)
+        return self.builder.call(fn, [e0], name)
+
+# ensure lifetime of the execution engine persists (else call to compiled
+#   function will seg fault)
+exe_engines = []
+
+# ensure names for generated functions are unique
+link_names = set()
+current_link_suffix = 0
+
+
+def llvm_jit_code(expr, func_args=None):
+    """Create a native code function from a Sympy expression"""
+
+    global exe_engines, current_link_suffix
+    module = ll.Module('mod1')
+    lj = LLVMJitPrinter(func_args=func_args)
+
+    fp_type = ll.DoubleType()
+    arg_types = []
+    for arg in func_args:
+        arg_type = fp_type
+        arg_types.append(arg_type)
+
+    default_link_name = 'jit_func'
+    current_link_suffix += 1
+    link_name = default_link_name + str(current_link_suffix)
+    link_names.add(link_name)
+
+    fn_type = ll.FunctionType(fp_type, arg_types)
+    fn = ll.Function(module, fn_type, name=link_name)
+    lj.module = module
+    lj.param_dict = {}
+    for i, a in enumerate(func_args):
+        name = str(a)
+        fn.args[i].name = name
+        lj.param_dict[name] = fn.args[i]
+    bb_entry = fn.append_basic_block('entry')
+
+    lj.builder = ll.IRBuilder(bb_entry)
+    lj.fn = fn
+
+    ret = lj._print(expr)
+    lj.builder.ret(ret)
+
+    strmod = str(module)
+    if False:
+        print("LLVM IR")
+        print(strmod)
+
+    llmod = llvm.parse_assembly(strmod)
+
+    pmb = llvm.create_pass_manager_builder()
+    pmb.opt_level = 2
+    pass_manager = llvm.create_module_pass_manager()
+    pmb.populate(pass_manager)
+
+    pass_manager.run(llmod)
+
+    target_machine = llvm.Target.from_default_triple().create_target_machine()
+    exe_eng = llvm.create_mcjit_compiler(llmod, target_machine)
+    exe_eng.finalize_object()
+    exe_engines.append(exe_eng)
+
+    if False:
+        print("Assembly")
+        print(target_machine.emit_assembly(llmod))
+
+    fptr = exe_eng.get_pointer_to_function(llmod.get_function(link_name))
+
+    return fptr
+
+
+@doctest_depends_on(modules=('llvmlite',))
+def get_jit_callable(expr, func_args=None):
```

Comment:
`get_llvm_callable` seems like a better name. LLVM and JIT are not interchangeable. There are other types of JIT than LLVM.